R-packages/evalcast/vignettes/intro-evalcast.Rmd

```diff
@@ -44,87 +48,142 @@ library(evalcast)
 library(covidcast)
 library(lubridate)
 
-forecast_dates <- get_forecast_dates("CMU-TimeSeries")
-forecast_dates_july <- forecast_dates[forecast_dates >= "2020-07-01" & forecast_dates <= "2020-07-31"]
+forecast_dates <- get_covidhub_forecast_dates("CMU-TimeSeries")
+forecast_dates_july <- forecast_dates[forecast_dates >= "2020-07-01" & 
+                                        forecast_dates <= "2020-07-31"]
 
 # Retrieve past predictions from CovidHub
-predictions_cards_yyg <- get_covidhub_predictions("YYG-ParamSearch", forecast_dates_july)
-predictions_cards_cmu <- get_covidhub_predictions("CMU-TimeSeries", forecast_dates_july)
-
+predictions_cards_ens <- get_covidhub_predictions(
+  "COVIDhub-ensemble", forecast_dates_july, ahead = 3, 
+  signal = "deaths_incidence_num") %>%
+  filter(nchar(geo_value)==2) # remove counties
+predictions_cards_cmu <- get_covidhub_predictions(
+  "CMU-TimeSeries", forecast_dates_july,  ahead = 3, 
+  signal = "deaths_incidence_num")
 # Make new predictions using baseline_forecaster
 predictions_cards <- get_predictions(baseline_forecaster,
                                      name_of_forecaster = "baseline",
                                      signals = signals,
                                      forecast_dates = forecast_dates_july,
                                      incidence_period = "epiweek",
                                      ahead = 3,
+                                     signal_aggregation = "long",
                                      geo_type = "state")
 ```
 
-`baseline_forecaster` created a list of `r length(predictions_cards)` predictions cards, (one for each `(forecast date, ahead)` pair).  Let's look at a predictions card:
+`get_predictions()` and `get_covidhub_predictions()` return a long `data.frame` with `r nrow(predictions_cards)` rows, (one for each `(forecast date, ahead, geo_value, quantile)` combination).
 
 ```{r}
-predictions_cards[[1]]
+predictions_cards_cmu
 ```
 
-Forecasts are probabilistic, meaning that rather than predicting a single value, a distribution of values is generated. For example, the baseline forecaster would have predicted the following distribution in California for `r attributes(predictions_cards_cmu[[1]])$ahead` week(s) after `r attributes(predictions_cards_cmu[[1]])$forecast_date`
+
+Let's look one location:
+
 ```{r}
-pc <- predictions_cards_cmu[[1]]
-ca_fips = covidcast::name_to_fips("California")
-ca_code = substr(ca_fips, 0, 2) # States are identified by a 2 character code
-pc[pc$location == ca_code, ]$forecast_distribution[[1]]
+predictions_cards_cmu %>% 
+  filter(geo_value == "mi", forecast_date == forecast_dates_july[1]) %>%
```

Comment:
Can we still use `name_to_fips`? I imagine some users (particularly international ones) won't immediately recognize US postal abbreviations.

Reply:
I removed the codes for states completely. The reason is that covidcast uses lowercase postal codes. So the goal is to be consistent with that. If you really want to convert, you can, but all the state-level data throughout the package uses letters. (This also avoids a bug in `fips_to_name(c("01", "10"))` which returns Alabama twice instead of Alabama and Delaware.

Reply:
Ah sorry, I meant `name_to_abbr` instead of `name_to_fips`. Made the change, I'll leave the comment open for you to object or resolve.